modules/api/src/main/java/com/intuit/wasabi/api/ApplicationsResource.java

```diff
@@ -373,4 +376,48 @@ public Response getPagesAndAssociatedExperimentsForApplication(
             throw exception;
         }
     }
+
+    /**
+     * Returns a Map of allowed Applications to the corresponding
+     * tags that have been stored with them.
+     *
+     * @param authorizationHeader the authorization headers
+     * @return Response object containing the Applications with their tags
+     */
+    @GET
+    @Path("/tags")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Returns a Map of Applications to their tags",
+            response = Map.class)
+    @Timed
+    public Response getAllExperimentTags(
+            @HeaderParam(AUTHORIZATION)
+            @ApiParam(value = EXAMPLE_AUTHORIZATION_HEADER, required = true)
+            final String authorizationHeader) {
+        try {
+
+            // get the for the user visible Applications
+            List<Application.Name> allApplications = experiments.getApplications();
+            Set<Application.Name> allowed = new HashSet<>();
+            UserInfo.Username userName = authorization.getUser(authorizationHeader);
+
+            for (Application.Name applicationName : allApplications) {
```

Comment:
@abhishekjain88 Agree, it would be smart to use getUserPermissionsList() first to get the list of user permissions (Applications & roles in those applications) and then get the tags for only applicable applications...